# 1. Cleaning

### importing data 

In [2]:
import pandas as pd
import openpyxl

In [3]:
csv_file = "D:\\ironhack\\proyectos\\SharkAttack_proy1\DATA\\attacks_1.csv"
shark_attack = pd.read_csv(csv_file, encoding="cp1252")

In [4]:
shark_attack

Case Number         Date    Year        Type    Country  \
0      2018.06.25  25-Jun-2018  2018.0     Boating        USA   
1      2018.06.18  18-Jun-2018  2018.0  Unprovoked        USA   
2      2018.06.09  09-Jun-2018  2018.0     Invalid        USA   
3      2018.06.08  08-Jun-2018  2018.0  Unprovoked  AUSTRALIA   
4      2018.06.04  04-Jun-2018  2018.0    Provoked     MEXICO   
...           ...          ...     ...         ...        ...   
25718         NaN          NaN     NaN         NaN        NaN   
25719         NaN          NaN     NaN         NaN        NaN   
25720         NaN          NaN     NaN         NaN        NaN   
25721         NaN          NaN     NaN         NaN        NaN   
25722          xx          NaN     NaN         NaN        NaN   

                  Area                        Location     Activity  \
0           California     Oceanside, San Diego County     Paddling   
1              Georgia  St. Simon Island, Glynn County     Standing   
2               Hawaii                    Habush, Oahu      Surfing   
3      New South Wales              Arrawarra Headland      Surfing   
4               Colima                        La Ticla  Free diving   
...                ...                             ...          ...   
25718              NaN                             NaN          NaN   
25719              NaN                             NaN          NaN   
25720              NaN                             NaN          NaN   
25721              NaN                             NaN          NaN   
25722              NaN                             NaN          NaN   

                  Name Sex   ...         Species   \
0          Julie Wolfe    F  ...      White shark   
1      Adyson McNeely     F  ...              NaN   
2          John Denges    M  ...              NaN   
3                 male    M  ...        2 m shark   
4       Gustavo Ramos     M  ...  Tiger shark, 3m   
...                ...  ...  ...              ...   
25718              NaN  NaN  ...              NaN   
25719              NaN  NaN  ...              NaN   
25720              NaN  NaN  ...              NaN   
25721              NaN  NaN  ...              NaN   
25722              NaN  NaN  ...              NaN   

               Investigator or Source                       pdf  \
0                    R. Collier, GSAF      2018.06.25-Wolfe.pdf   
1      K.McMurray, TrackingSharks.com    2018.06.18-McNeely.pdf   
2      K.McMurray, TrackingSharks.com     2018.06.09-Denges.pdf   
3                      B. Myatt, GSAF  2018.06.08-Arrawarra.pdf   
4                           A .Kipper      2018.06.04-Ramos.pdf   
...                               ...                       ...   
25718                             NaN                       NaN   
25719                             NaN                       NaN   
25720                             NaN                       NaN   
25721                             NaN                       NaN   
25722                             NaN                       NaN   

                                            href formula  \
0      http://sharkattackfile.net/spreadsheets/pdf_di...   
1      http://sharkattackfile.net/spreadsheets/pdf_di...   
2      http://sharkattackfile.net/spreadsheets/pdf_di...   
3      http://sharkattackfile.net/spreadsheets/pdf_di...   
4      http://sharkattackfile.net/spreadsheets/pdf_di...   
...                                                  ...   
25718                                                NaN   
25719                                                NaN   
25720                                                NaN   
25721                                                NaN   
25722                                                NaN   

                                                    href Case Number.1  \
0      http://sharkattackfile.net/spreadsheets/pdf_di...    2018.06.25   
1      http://sharkattackfile.net/spreadsheets/pdf_di...    2

In [5]:
shark_attack.sample(3)

Case Number         Date    Year      Type Country    Area  \
12849         NaN          NaN     NaN       NaN     NaN     NaN   
18772         NaN          NaN     NaN       NaN     NaN     NaN   
5017   1931.06.13  13-Jun-1931  1931.0  Provoked     USA  Hawaii   

                  Location                                           Activity  \
12849                  NaN                                                NaN   
18772                  NaN                                                NaN   
5017   Pearl Harbor, O'ahu  Gaffing & attempting to bring onboard a harpoo...   

                        Name Sex   ...                Species   \
12849                    NaN  NaN  ...                     NaN   
18772                    NaN  NaN  ...                     NaN   
5017   Lieutenant Williamson    M  ...  Tiger shark, 3 m [10']   

                                  Investigator or Source  \
12849                                                NaN   
18772                                                NaN   
5017   G.H. Balazs; J. Borg, p.70; L. Taylor (1993), ...   

                             pdf  \
12849                        NaN   
18772                        NaN   
5017   1931.06.13-Williamson.pdf   

                                            href formula  \
12849                                                NaN   
18772                                                NaN   
5017   http://sharkattackfile.net/spreadsheets/pdf_di...   

                                                    href Case Number.1  \
12849                                                NaN           NaN   
18772                                                NaN           NaN   
5017   http://sharkattackfile.net/spreadsheets/pdf_di...    1931.06.13   

      Case Number.2 original order Unnamed: 22 Unnamed: 23  
12849           NaN            NaN         NaN         NaN  
18772           NaN            NaN         NaN         NaN  
5017     1931.06.13         1286.0         NaN         NaN  

[3 rows x 24 columns]

In [6]:
shark_attack.shape[0]

25723

In [7]:
shark_attack.dtypes

Case Number                object
Date                       object
Year                      float64
Type                       object
Country                    object
Area                       object
Location                   object
Activity                   object
Name                       object
Sex                        object
Age                        object
Injury                     object
Fatal (Y/N)                object
Time                       object
Species                    object
Investigator or Source     object
pdf                        object
href formula               object
href                       object
Case Number.1              object
Case Number.2              object
original order            float64
Unnamed: 22                object
Unnamed: 23                object
dtype: object

### Null values

In [8]:
shark_attack.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 25723 entries, 0 to 25722
Data columns (total 24 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Case Number             8702 non-null   object 
 1   Date                    6302 non-null   object 
 2   Year                    6300 non-null   float64
 3   Type                    6298 non-null   object 
 4   Country                 6252 non-null   object 
 5   Area                    5847 non-null   object 
 6   Location                5762 non-null   object 
 7   Activity                5758 non-null   object 
 8   Name                    6092 non-null   object 
 9   Sex                     5737 non-null   object 
 10  Age                     3471 non-null   object 
 11  Injury                  6274 non-null   object 
 12  Fatal (Y/N)             5763 non-null   object 
 13  Time                    2948 non-null   object 
 14  Species                 3464 non-null 

In [9]:
#revisar si necesito eliminar los NaN
#puedo meterme directo a explorar la columna a ver como se ven con los NaN
#llamando el key y poniendole .isna()

In [10]:
#pd.isnull(shark_attack).sum() same formula
shark_attack.isna().sum()
#for NaN, un promedio de 18000NaNs en una lista de 19803
#conclucion, demasiados NaN tendre que eliminar

Case Number               17021
Date                      19421
Year                      19423
Type                      19425
Country                   19471
Area                      19876
Location                  19961
Activity                  19965
Name                      19631
Sex                       19986
Age                       22252
Injury                    19449
Fatal (Y/N)               19960
Time                      22775
Species                   22259
Investigator or Source    19438
pdf                       19421
href formula              19422
href                      19421
Case Number.1             19421
Case Number.2             19421
original order            19414
Unnamed: 22               25722
Unnamed: 23               25721
dtype: int64

In [11]:
shark_attack.notnull().sum()

Case Number               8702
Date                      6302
Year                      6300
Type                      6298
Country                   6252
Area                      5847
Location                  5762
Activity                  5758
Name                      6092
Sex                       5737
Age                       3471
Injury                    6274
Fatal (Y/N)               5763
Time                      2948
Species                   3464
Investigator or Source    6285
pdf                       6302
href formula              6301
href                      6302
Case Number.1             6302
Case Number.2             6302
original order            6309
Unnamed: 22                  1
Unnamed: 23                  2
dtype: int64

After this filter we can see that it has tons of null data, from 25723 original rows we can see that it will let us with almos 6,000 rows. Not much information and we can clean it.


next step it will be to drop (errase) the rows that dont apport nothing to the data. 
If we throw the formula od dropna with the fuction "any" its not an option, every single row has even one missed cell, we will be applying the function "all"

In [12]:
shark_attack.dropna(how="all",inplace=True)
shark_attack

Case Number         Date    Year        Type    Country  \
0      2018.06.25  25-Jun-2018  2018.0     Boating        USA   
1      2018.06.18  18-Jun-2018  2018.0  Unprovoked        USA   
2      2018.06.09  09-Jun-2018  2018.0     Invalid        USA   
3      2018.06.08  08-Jun-2018  2018.0  Unprovoked  AUSTRALIA   
4      2018.06.04  04-Jun-2018  2018.0    Provoked     MEXICO   
...           ...          ...     ...         ...        ...   
8698            0          NaN     NaN         NaN        NaN   
8699            0          NaN     NaN         NaN        NaN   
8700            0          NaN     NaN         NaN        NaN   
8701            0          NaN     NaN         NaN        NaN   
25722          xx          NaN     NaN         NaN        NaN   

                  Area                        Location     Activity  \
0           California     Oceanside, San Diego County     Paddling   
1              Georgia  St. Simon Island, Glynn County     Standing   
2               Hawaii                    Habush, Oahu      Surfing   
3      New South Wales              Arrawarra Headland      Surfing   
4               Colima                        La Ticla  Free diving   
...                ...                             ...          ...   
8698               NaN                             NaN          NaN   
8699               NaN                             NaN          NaN   
8700               NaN                             NaN          NaN   
8701               NaN                             NaN          NaN   
25722              NaN                             NaN          NaN   

                  Name Sex   ...         Species   \
0          Julie Wolfe    F  ...      White shark   
1      Adyson McNeely     F  ...              NaN   
2          John Denges    M  ...              NaN   
3                 male    M  ...        2 m shark   
4       Gustavo Ramos     M  ...  Tiger shark, 3m   
...                ...  ...  ...              ...   
8698               NaN  NaN  ...              NaN   
8699               NaN  NaN  ...              NaN   
8700               NaN  NaN  ...              NaN   
8701               NaN  NaN  ...              NaN   
25722              NaN  NaN  ...              NaN   

               Investigator or Source                       pdf  \
0                    R. Collier, GSAF      2018.06.25-Wolfe.pdf   
1      K.McMurray, TrackingSharks.com    2018.06.18-McNeely.pdf   
2      K.McMurray, TrackingSharks.com     2018.06.09-Denges.pdf   
3                      B. Myatt, GSAF  2018.06.08-Arrawarra.pdf   
4                           A .Kipper      2018.06.04-Ramos.pdf   
...                               ...                       ...   
8698                              NaN                       NaN   
8699                              NaN                       NaN   
8700                              NaN                       NaN   
8701                              NaN                       NaN   
25722                             NaN                       NaN   

                                            href formula  \
0      http://sharkattackfile.net/spreadsheets/pdf_di...   
1      http://sharkattackfile.net/spreadsheets/pdf_di...   
2      http://sharkattackfile.net/spreadsheets/pdf_di...   
3      http://sharkattackfile.net/spreadsheets/pdf_di...   
4      http://sharkattackfile.net/spreadsheets/pdf_di...   
...                                                  ...   
8698                                                 NaN   
8699                                                 NaN   
8700                                                 NaN   
8701                                                 NaN   
25722                                                NaN   

                                                    href Case Number.1  \
0      http://sharkattackfile.net/spreadsheets/pdf_di...    2018.06.25   
1      http://sharkattackfile.net/spreadsheets/pdf_di...    2

In [13]:
shark_attack.shape[0]

8703

it let us with only 8703 rows. But we still have some missing values in the cells thats wy apears in the formula below that we still have a lot of them, probably half of the columns have NANs.

In [14]:
shark_attack.isna().sum()

Case Number                  1
Date                      2401
Year                      2403
Type                      2405
Country                   2451
Area                      2856
Location                  2941
Activity                  2945
Name                      2611
Sex                       2966
Age                       5232
Injury                    2429
Fatal (Y/N)               2940
Time                      5755
Species                   5239
Investigator or Source    2418
pdf                       2401
href formula              2402
href                      2401
Case Number.1             2401
Case Number.2             2401
original order            2394
Unnamed: 22               8702
Unnamed: 23               8701
dtype: int64

### Duplicated

In [15]:
shark_attack.shape[0]

8703

In [16]:
shark_attack.duplicated().sum()

2392

This tell us that we have 2,392 raws that are duplicated and will have to be removed, otherwise it will leave us with some false information.

In [17]:
shark_attack.drop_duplicates(inplace=True)
shark_attack.sample(5)

Case Number          Date    Year          Type      Country  \
840     2011.12.02   02-Dec-2011  2011.0    Unprovoked    AUSTRALIA   
3030    1981.05.16   16-May-1981  1981.0    Unprovoked          USA   
5565    1899.11.20   20-Nov-1899  1899.0  Sea Disaster  PHILIPPINES   
626   2013.09.07.a   07-Sep-2013  2013.0    Unprovoked          USA   
3367  1971.04.05.a    5-Apr-1971  1971.0    Unprovoked    VENEZUELA   

                    Area                           Location      Activity  \
840      New South Wales                        Broken Head       Surfing   
3030             Florida         Jacksonville, Duval County        Diving   
5565  Mindoro Occidental                  Off Lubang Island  Sea Disaster   
626              Florida  New Smyrna Beach,  Volusia County      Standing   
3367                 NaN                                NaN           NaN   

                              Name Sex   ...     Species   \
840                  Milton Carter    M  ...          NaN   
3030                    Carey Ford    M  ...  Lemon shark   
5565  wreck of the steamship Hubeh  NaN  ...          NaN   
626          Marco Edmundo Cardiel    M  ...     3' shark   
3367               Tarcisio Mayora    M  ...          NaN   

                    Investigator or Source                           pdf  \
840               Northern Star, 12/3/2011         2011.12.02-Carter.pdf   
3030                         E. Pace, FSAF        1981.05.16-NV-Ford.pdf   
5565                 The Mercury, 1/191900    1899.11.20-Hupeh-wreck.pdf   
626             Orlando Sentinel, 9/7/2013      2013.09.07.a-Cardiel.pdf   
3367  H.D. Baldridge (1994) SAF Case #1646  1971.04.05.a-NV-T-Mayora.pdf   

                                           href formula  \
840   http://sharkattackfile.net/spreadsheets/pdf_di...   
3030  http://sharkattackfile.net/spreadsheets/pdf_di...   
5565  http://sharkattackfile.net/spreadsheets/pdf_di...   
626   http://sharkattackfile.net/spreadsheets/pdf_di...   
3367  http://sharkattackfile.net/spreadsheets/pdf_di...   

                                                   href Case Number.1  \
840   http://sharkattackfile.net/spreadsheets/pdf_di...    2011.12.02   
3030  http://sharkattackfile.net/spreadsheets/pdf_di...    1981.05.16   
5565  http://sharkattackfile.net/spreadsheets/pdf_di...    1899.11.20   
626   http://sharkattackfile.net/spreadsheets/pdf_di...  2013.09.07.a   
3367  http://sharkattackfile.net/spreadsheets/pdf_di...  1971.04.05.a   

     Case Number.2 original order Unnamed: 22 Unnamed: 23  
840     2011.12.02         5463.0         NaN         NaN  
3030    1981.05.16         3273.0         NaN         NaN  
5565    1899.11.20          738.0         NaN         NaN  
626   2013.09.07.a         5677.0         NaN         NaN  
3367  1971.04.05.a         2936.0         NaN         NaN  

[5 rows x 24 columns]

In [18]:
shark_attack.shape[0]

6311

This leave us with only 6,311 rows that are not duplicated and have minimum 1 real value.
Because I can notice that there are still some rows that looks preatty empty and the few values that could have it might not be important to us and our investigation.

### columns

passing throw the Data frame I notice that theres some columns that are not relevant to our investigation, that doesnt apport anything of value that we could could benefit

In [19]:
shark_attack.sample()

Case Number         Date    Year        Type Country     Area  \
3905  1961.04.16.b  16-Apr-1961  1961.0  Unprovoked     USA  Florida   

                     Location      Activity          Name Sex   ...  \
3905  Fowey Rock Light, Miami  Spearfishing  Rolf Ericson    M  ...   

                    Species   \
3905  Tiger shark, 3 m [10']   

                                 Investigator or Source  \
3905  W. M. Stephens; H.D. Baldridge, p.166; Clark, ...   

                           pdf  \
3905  1961.04.16.b-Ericson.pdf   

                                           href formula  \
3905  http://sharkattackfile.net/spreadsheets/pdf_di...   

                                                   href Case Number.1  \
3905  http://sharkattackfile.net/spreadsheets/pdf_di...  1961.04.16.b   

     Case Number.2 original order Unnamed: 22 Unnamed: 23  
3905  1961.04.16.b         2398.0         NaN         NaN  

[1 rows x 24 columns]

In [20]:
shark_attack.columns

Index(['Case Number', 'Date', 'Year', 'Type', 'Country', 'Area', 'Location',
       'Activity', 'Name', 'Sex ', 'Age', 'Injury', 'Fatal (Y/N)', 'Time',
       'Species ', 'Investigator or Source', 'pdf', 'href formula', 'href',
       'Case Number.1', 'Case Number.2', 'original order', 'Unnamed: 22',
       'Unnamed: 23'],
      dtype='object')

There are some columns that makes me have some interest, in wich ones we could errase because of our lak of information for us, for example the las 2 columns 'Unnamed: 22', 'Unnamed: 23', seen the samples we could see that theres only non values and actually the tittle doesn´t give us any explanetion.
We will erase them.

In [21]:
shark_attack[['Unnamed: 22', "Unnamed: 23"]].sample(5)

Unnamed: 22 Unnamed: 23
3787         NaN         NaN
1505         NaN         NaN
2884         NaN         NaN
260          NaN         NaN
4416         NaN         NaN

In [22]:
shark_attack.drop(columns =['Unnamed: 22', "Unnamed: 23"], inplace=True, axis=1)

In [33]:
shark_attack.sample()

Case Number         Date    Year     Type Country            Area  \
2358  1995.08.13.a  13-Aug-1995  1995.0  Invalid     USA  North Carolina   

                           Location  Activity          Name Sex   ...  \
2358  Emerald Isle, Carteret County  Swimming  Kyle Dickens    M  ...   

     Fatal (Y/N) Time Species  Investigator or Source  \
2358         NaN  NaN  Invalid    J. L. Almeida, M.D.   

                           pdf  \
2358  1995.08.13.a-Dickens.pdf   

                                           href formula  \
2358  http://sharkattackfile.net/spreadsheets/pdf_di...   

                                                   href Case Number.1  \
2358  http://sharkattackfile.net/spreadsheets/pdf_di...  1995.08.13.a   

     Case Number.2 original order  
2358  1995.08.13.a         3945.0  

[1 rows x 22 columns]

more research in columns: lets see if the internet links could help us with our research.

In [28]:
shark_attack['href formula'].sample(5)                        

4321    http://sharkattackfile.net/spreadsheets/pdf_di...
1871    http://sharkattackfile.net/spreadsheets/pdf_di...
5605    http://sharkattackfile.net/spreadsheets/pdf_di...
5206    http://sharkattackfile.net/spreadsheets/pdf_di...
1903    http://sharkattackfile.net/spreadsheets/pdf_di...
Name: href formula, dtype: object

In [29]:
shark_attack['href'].sample(5) 

6232    http://sharkattackfile.net/spreadsheets/pdf_di...
3485    http://sharkattackfile.net/spreadsheets/pdf_di...
638     http://sharkattackfile.net/spreadsheets/pdf_di...
5127    http://sharkattackfile.net/spreadsheets/pdf_di...
1583    http://sharkattackfile.net/spreadsheets/pdf_di...
Name: href, dtype: object

Exploring this 2 columns 'href formula' and 'href' we can realice that they apporr some internet links, but going throw them they the pages dont exist amymore.

In [30]:
shark_attack['Case Number.1'].sample(5) 

4174    1958.00.00.h
2370      1995.07.23
6012    1853.00.00.b
2452      1994.02.00
4351    1954.00.00.e
Name: Case Number.1, dtype: object

other columns that dont apport anything of value are 'Case Number.1', 'Case Number.2'  we can see that the information there is actually the same and its the date of the attack that we do have it already in other columns at the beginning, and 'original order' dont aport us anything.
we´ll pass to some more drop cleaning.

In [35]:
shark_attack.drop(columns =['href formula', 'href', 'Case Number.1', 'Case Number.2', 'original order'], inplace=True, axis=1)

In [43]:
shark_attack.sample(50)

Case Number                       Date    Year          Type  \
2125      1999.04.22                22-Apr-1999  1999.0    Unprovoked   
2948      1983.03.22                22-Mar-1983  1983.0    Unprovoked   
2391      1995.04.13                13-Apr-1995  1995.0    Unprovoked   
5956      1863.03.05                05-Mar-1863  1863.0    Unprovoked   
5329      1915.00.00                   Ca. 1915  1915.0       Invalid   
4522    1949.00.00.g                  1949-1950  1949.0       Boating   
287       2016.04.13                13-Apr-2016  2016.0    Unprovoked   
6167    1617.00.00.R              Reported 1617  1617.0    Unprovoked   
1813      2002.11.02                02-Nov-2002  2002.0    Unprovoked   
5939    1864.09.18.R       Reported 18-Sep-1864  1864.0      Provoked   
3853      1962.01.08                08-Jan-1962  1962.0       Invalid   
2347    1995.08.31.b                31-Aug-1995  1995.0    Unprovoked   
2743      1988.03.31                31-Mar-1988  1988.0    Unprovoked   
1047      2010.01.30                31-Jan-2010  2010.0    Unprovoked   
2576      1991.11.19                19-Nov-1991  1991.0    Unprovoked   
3285      1973.12.25                25-Dec-1973  1973.0    Unprovoked   
1544    2005.09.27.R  Reported      27-Sep-2005  2005.0    Unprovoked   
4826  1938.00.00.e.R              Reported 1938  1938.0    Unprovoked   
5809      1881.08.12                12-Aug-1881  1881.0    Unprovoked   
4880      1936.04.22                22-Apr-1936  1936.0    Unprovoked   
4603      1946.02.16                16-Feb-1946  1946.0    Unprovoked   
5754      1886.05.06                06-May-1886  1886.0    Unprovoked   
1145    2009.03.19.a                19-Mar-2009  2009.0    Unprovoked   
100       2017.08.23                23-Aug-2017  2017.0    Unprovoked   
1819    2002.10.05.a                05-Oct-2002  2002.0    Unprovoked   
424       2015.05.15                15-May-2015  2015.0    Unprovoked   
4833    1937.11.06.R       Reported 06-Nov-1937  1937.0  Sea Disaster   
3407      1970.01.00                   Jan-1970  1970.0    Unprovoked   
5291      1917.05.05       Reported 05-May-1917  1917.0    Unprovoked   
3105      1979.03.00                   Mar-1979  1979.0    Unprovoked   
3325      1972.09.04                04-Sep-1972  1972.0       Invalid   
1661    2004.06.26.a                26-Jun-2004  2004.0       Invalid   
304       2016.02.19                19-Feb-2016  2016.0    Unprovoked   
1219      2008.08.18                18-Aug-2008  2008.0       Invalid   
6080    1836.07.26.R               1836.07.26.R  1836.0       Invalid   
6056    1845.00.00.R              Reported 1845  1845.0    Unprovoked   
2997    1982.00.00.b                       1982  1982.0          Boat   
1235      2008.07.13                13-Jul-2008  2008.0       Invalid   
1126    2009.05.16.b                16-May-2009  2009.0    Unprovoked   
3138      1977.10.30                30-Oct-1977  1977.0    Unprovoked   
4626      1945.02.05                05-Feb-1945  1945.0    Unprovoked   
4967      1933.05.03                03-May-1933  1933.0  Sea Disaster   
3113      1978.09.27                27-Sep-1978  1978.0    Unprovoked   
5603    1898.00.00.a             Summer of 1898  1898.0    Unprovoked   
686       2013.04.10                10-Apr-2013  2013.0    Unprovoked   
20      2018.04.25.b                25-Apr-2018  2018.0  Questionable   
4715      1942.03.08                08-Mar-1942  1942.0  Sea Disaster   
3966      1960.08.30                30-Aug-1960  1960.0    Unprovoked   
1227    2008.07.26.a                26-Jul-2008  2008.0    Unprovoked   
5110    1928.03.28.R       Reported 28-Mar-1928  1928.0    Unprovoked   

               Country                   Area  \
2125         MAURITIUS             Grand Baie   
2948      SOUTH AFRICA  Eastern Cape Province   
2391               USA                Florida   
5956         AUSTRALIA               Victoria   
5329               USA                Fl

In [48]:
shark_attack["Type"].value_counts()

Unprovoked      4595
Provoked         574
Invalid          547
Sea Disaster     239
Boating          203
Boat             137
Questionable       2
Boatomg            1
Name: Type, dtype: int64

In [50]:
shark_attack['Sex '].value_counts()

M      5094
F       637
M         2
N         2
lli       1
.         1
Name: Sex , dtype: int64

In [51]:
shark_attack['Age'].value_counts()

17                154
18                150
19                142
20                141
15                139
                 ... 
20?                 1
 28                 1
7      &    31      1
 30                 1
13 or 14            1
Name: Age, Length: 157, dtype: int64

In [45]:
#no proprciona mucha informacion, o nula.
shark_attack.describe()

Year
count  6300.000000
mean   1927.272381
std     281.116308
min       0.000000
25%    1942.000000
50%    1977.000000
75%    2005.000000
max    2018.000000

In [46]:
shark_attack.sample()

Case Number         Date    Year        Type Country        Area  \
3995  1960.04.24  24-Apr-1960  1960.0  Unprovoked     USA  California   

                         Location                                 Activity  \
3995  Tomales Point, Marin County  Skindiving for abalone (but at surface)   

                  Name Sex  Age                   Injury Fatal (Y/N)   Time  \
3995  Frank I. Gilbert    M  48  Foot & swim fin bitten            N  14h00   

                       Species   \
3995  White shark, 4.9 m [16']    

                                 Investigator or Source  \
3995  F.I. Gilbert; Follett, p. 192-198; D. Miller &...   

                                 pdf  
3995  1960.04.24-Gilbert_Collier.pdf

In [47]:
shark_attack["Country"].value_counts()

USA                       2229
AUSTRALIA                 1338
SOUTH AFRICA               579
PAPUA NEW GUINEA           134
NEW ZEALAND                128
                          ... 
MALDIVE ISLANDS              1
NICARAGUA                    1
NORTH SEA                    1
RED SEA / INDIAN OCEAN       1
CEYLON (SRI LANKA)           1
Name: Country, Length: 212, dtype: int64